torchmetrics/utilities/checks.py

```diff
@@ -604,3 +605,117 @@ def _check_retrieval_target_and_prediction_types(
     preds = preds.float()
 
     return preds.flatten(), target.flatten()
+
+
+def _allclose_recursive(res1: Any, res2: Any, atol: float = 1e-8):
```

Comment:
can we add test functions for this helper only? Without the need for the checking function below?

Reply:
done